services/web/server/src/simcore_service_webserver/login/decorators.py

```diff
@@ -42,12 +44,23 @@ async def get_foo(request: web.Request):
     async def _wrapper(request: web.Request):
         """
         Raises:
-            HTTPUnauthorized: if request authorization check fails
+            HTTPUnauthorized: if unauthorized user
+            HTTPForbidden: if user not allowed in product
         """
         # WARNING: note that check_authorized is patched in some tests.
         # Careful when changing the function signature
-        request[RQT_USERID_KEY] = await check_authorized(request)
+        user_id = await check_user_authorized(request)
+
+        await check_user_permission(
```

Comment:
Q: do any of these calls hit the DB? If only one does I would put that one last.

Reply:
@GitHK They have to be in this order. 

The db is "protected" with at TTL-based cache. Check the test and there you can see how many times the db is hit and also the overhead that the auth does to handlers. This test will keep this in check in case we "de-tune" it accidentaly in the future.